lib/MusicBrainz/Script/RemoveExpiredSessions.pm

```diff
@@ -0,0 +1,102 @@
+package MusicBrainz::Script::RemoveExpiredSessions;
+
+=head1 DESCRIPTION
+
+This is a cleanup script related to MBS-8692. It removes all Catalyst
+"expires" session keys for which the expire time (stored as the value) is in
+the past. For the other session sub-keys used by Catalyst, namely "session"
+and "flash", cleanup is not necessary because they already had their Redis
+expire time set correctly.
+
+The base issue for "expires" keys is now also fixed, so it should not be
+necessary to re-run this script.
+
+=cut
+
+use Moose;
+use namespace::autoclean;
+
+with 'MooseX::Runnable';
+with 'MooseX::Getopt';
+with 'MusicBrainz::Script::Role::Context';
+
+has 'verbose' => (
+    isa => 'Bool',
+    is => 'ro',
+    default => sub { -t },
+);
+
+has 'dry_run' => (
+    isa => 'Bool',
+    is => 'ro',
+    default => 0,
+    traits => [ 'Getopt' ],
+    cmd_flag => 'dry-run',
+);
+
+sub run {
+    my ($self, @args) = @_;
+    die "Usage error ($0 takes no arguments)" if @args;
+
+    my $r = $self->c->redis;
+    my $now = time + 5; # five seconds grace period for server differences
+
+    printf "Fetching entries from database; prefix used is \"%s\".\n", $r->prefix if $self->verbose;
+    my @keys = $r->_connection->keys($r->prefix . "expires:*");
+        # KEYS is very heavy, but our current Redis doesn't have SCAN
```

Comment:
I had originally written the script to use [SCAN](http://redis.io/commands/scan), which would be a perfect fit here. Then I made a test run and noticed the command is unknown in our Redis version …

So this will read 4.8 million keys in one go. Let’s see if it works. ;-) (6000 keys were no problem in a test run.)